### PR TITLE
[Model Monitoring] Fix SQL store with multiple filter criteria

### PR DESCRIPTION
**Pull request by jond01**
Fixes [ML-6541](https://iguazio.atlassian.net/browse/ML-6541).

Add a test that fails before the fix introduced here: `test_unique_last_analyzed_per_app`.
The bug comes from the fact that `sqlalchemy.sql.text` cannot combine by itself multiple conditions.

[ML-6541]: https://iguazio.atlassian.net/browse/ML-6541?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mlrun/model_monitoring/db/stores/sqldb/sql_store.py [hunk 1]
```diff
@@ -441,14 +439,15 @@ def update_last_analyzed(
         """
         self._init_monitoring_schedules_table()
 
-        application_filter_dict = self.filter_endpoint_and_application_name(
-            endpoint_id=endpoint_id, application_name=application_name
-        )
+        criteria = [
+            self.MonitoringSchedulesTable.endpoint_id == endpoint_id,
+            self.MonitoringSchedulesTable.application_name == application_name,
+        ]
```
Comment (davesh0812) [inline mlrun/model_monitoring/db/stores/sqldb/sql_store.py]: You are using this lines or part of them for all the tables a lot of times in your code, I think you can consider to create some helper method such that `_get_criteria(table_name, endpoint_id, application_name)` and to use it in all the relevant places for all the tables.

File: mlrun/model_monitoring/db/stores/sqldb/sql_store.py [hunk 2]
```diff
@@ -462,31 +461,31 @@ def update_last_analyzed(
...
+                self.MonitoringSchedulesTable.application_name == application_name
+            )
         # Delete the model endpoint record using sqlalchemy ORM
-        self._delete(table=self.MonitoringSchedulesTable, **application_filter_dict)
+        self._delete(table=self.MonitoringSchedulesTable, criteria=criteria)
 
-    def _delete_application_result(
-        self, endpoint_id: str = "", application_name: str = ""
-    ):
+    def _delete_application_result(self, endpoint_id: str) -> None:
```
Comment (davesh0812) [inline mlrun/model_monitoring/db/stores/sqldb/sql_store.py]: please revert the option to filter on `application_name` as well, as we spoke.

File: mlrun/model_monitoring/db/stores/sqldb/models/__init__.py [hunk 3]
```diff
@@ -21,29 +22,42 @@
...
-        return MySQLApplicationResultTable
-    return SQLiteApplicationResultTable
+_MYSQL_SCHEME = "mysql:"
 
 
-def _get_monitoring_schedules_table(
+def _get_sql_table(
+    *,
+    mysql_table: MySQLTableType,
+    sqlite_table: SQLiteTableType,
```
Comment (davesh0812) [inline mlrun/model_monitoring/db/stores/sqldb/models/__init__.py:34]: Nice!

**Review by davesh0812 — COMMENTED** [review]
Looks Good!
Few extra comments.

File: mlrun/model_monitoring/db/stores/sqldb/sql_store.py (hunk 1 above)
Comment (jond01) [reply mlrun/model_monitoring/db/stores/sqldb/sql_store.py]: Done in 90b4988

File: mlrun/model_monitoring/db/stores/sqldb/sql_store.py (hunk 2 above)
Comment (jond01) [reply mlrun/model_monitoring/db/stores/sqldb/sql_store.py]: Done in 5a8a160

**Review by davesh0812 — APPROVED** [review]
LGTM!
 If it is not a big effort please add test for  `_delete_application_result` and pass to it `application_name` - not mandatory for now.

**Pull request merged**